### PR TITLE
[PD-3492][PD-3593] Add title, created/updated tracking to PartnerAssessments

### PR DESCRIPTION
**Pull request by koonsies**
## What's this PR do?
Adds custom title, created (by, on) updated (by, on) handling to Partner Assessments.

## Screenshots (if appropriate)
image

## Where should the reviewer start?
* Run migrations.

## How should this be manually tested?
* Navigate to Partner Assessments
* Add a new assessment with a custom title.
* do `dkm shell` from your command line
* execute the following commands
```
import pprint
from mypartners.models import PartnerAssessment
pa = PartnerAssessment.objects.all().last()
pprint.pprint(pa.__dict__)
```
* Make sure those values look correct.

### CODE REVIEW

File: gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx [hunk 1]
```diff
@@ -7,17 +7,16 @@ import Subhead from './Header/Subhead';
 import AssessmentForm from './Form/AssessmentForm';
 import RecordContainer from './Records/RecordContainer';
 
-
 /* PartnerAssessmentApp */
 class PartnerAssessmentApp extends Component {
   render() {
-    const { assessment, dateRange, dispatch, loading} = this.props;
+    const {assessment, dispatch, loading} = this.props;
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx]: `dateRange` in the connect() is no longer used and can be removed.

File: gulp/src/sass/partner_assessment/components/_sub_head.scss [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
 
-
 .sub-head {
   display: flex;
-  justify-content: space-between;
-  align-items: center;
-  padding: 0 20px;
-}
+  flex-direction: column;
+  padding: 20px 10px 15px;
```
Comment (aljordan82) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: 20px padding seems a bit excessive.

Comment (koonsies) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: @iartemiev, you probably aren't getting notifications from this PR, so tagging you on these comments.

Comment (koonsies) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: Feel free to make any changes directly to this branch, @iartemiev

Comment (iartemiev) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: @aljordan82 '20px' is what Sara and I decided on, but I'm open to suggestions

File: gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx (hunk 1 above)
Comment (iartemiev) [inline gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx]: Removed

File: gulp/src/sass/partner_assessment/components/_sub_head.scss (hunk 2 above)
Comment (aljordan82) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: 0 top-padding look fine to me. 20px is a waste of real estate  especially on mobile

Comment (iartemiev) [inline gulp/src/sass/partner_assessment/components/_sub_head.scss]: Ok, I put it back to 0

**Pull request merged**